File: server/test/detect.test.js
'use strict';
const { test } = require('node:test');
const assert = require('node:assert');
const { endsWithQuestion, parseChoices } = require('../lib/detect');

test('endsWithQuestion is true when text ends with a question mark', () => {
  assert.equal(endsWithQuestion('Do you want to proceed?'), true);
});

test('endsWithQuestion ignores trailing whitespace/newlines', () => {
  assert.equal(endsWithQuestion('Proceed?  \n\t'), true);
});

test('endsWithQuestion is false without a trailing question mark', () => {
  assert.equal(endsWithQuestion('All done.'), false);
  assert.equal(endsWithQuestion('a? and more'), false);
  assert.equal(endsWithQuestion(''), false);
});

test('parseChoices reads a numbered menu with dot separators', () => {
  const opts = parseChoices('Pick one:\n1. Yes\n2. No, keep it');
  assert.deepEqual(opts, [
    { n: 1, label: 'Yes' },
    { n: 2, label: 'No, keep it' },
  ]);
});

test('parseChoices accepts ")" separators and leading indentation', () => {
  const opts = parseChoices('  1) Alpha\n  2) Beta');
  assert.deepEqual(opts, [
    { n: 1, label: 'Alpha' },
    { n: 2, label: 'Beta' },
  ]);
});

test('parseChoices ignores lines that are not options', () => {
  assert.deepEqual(parseChoices('3.14 is pi\nno number here\n1.no space'), []);
});

File: server-go/internal/screen/detect_test.go
package screen

import "testing"

func choicesEq(a []Choice, b []Choice) bool {
	if len(a) != len(b) {
		return false
	}
	for i := range a {
		if a[i] != b[i] {
			return false
		}
	}
	return true
}

func TestParseChoicesDot(t *testing.T) {
	got := ParseChoices("Pick one:\n1. Yes\n2. No, keep it")
	want := []Choice{{1, "Yes"}, {2, "No, keep it"}}
	if !choicesEq(got, want) {
		t.Fatalf("got %+v", got)
	}
}

func TestParseChoicesParenIndent(t *testing.T) {
	got := ParseChoices("  1) Alpha\n  2) Beta")
	want := []Choice{{1, "Alpha"}, {2, "Beta"}}
	if !choicesEq(got, want) {
		t.Fatalf("got %+v", got)
	}
}

func TestParseChoicesIgnoresNonOptions(t *testing.T) {
	if got := ParseChoices("3.14 is pi\nno number here\n1.no space"); len(got) != 0 {
		t.Fatalf("got %+v", got)
	}
}
